flytekit/core/data_persistence.py

```diff
@@ -248,7 +248,10 @@ def get(self, from_path: str, to_path: str, recursive: bool = False, **kwargs):
                     self.strip_file_header(from_path), self.strip_file_header(to_path), dirs_exist_ok=True
                 )
             print(f"Getting {from_path} to {to_path}")
-            return file_system.get(from_path, to_path, recursive=recursive, **kwargs)
+            dst = file_system.get(from_path, to_path, recursive=recursive, **kwargs)
+            if isinstance(dst, (str, pathlib.Path)):
```

Comment:
IIUC, `[None]` is a guard value returned by `fsspec`, right? If that's the case, why don't we handle those explicitly?

Reply:
oh really?  didn't see that.  link me?

Reply:
and yes we should, but the action should still be to return the to_path.

Reply:
I don't think it's part of the spec and I fear this is specific to `s3fs`. Doing what we're doing in the PR is probably safer (as it might apply to other fsspec-compliant implementations).

Reply:
AsyncFileSystem.put or AsyncFileSystem.get always return a list.
- https://github.com/fsspec/filesystem_spec/blob/294248578994de67f8da00692de6872f1352ce33/fsspec/asyn.py#L574
- https://github.com/fsspec/filesystem_spec/blob/294248578994de67f8da00692de6872f1352ce33/fsspec/asyn.py#L242-L256